### PR TITLE
added python, PHP and javascript sample codes for API:Stashedit

### PR DESCRIPTION
**Pull request by zainababubakar**
Link to revised API documentation page in my Sandbox - https://www.mediawiki.org/wiki/User:Zaycodes/Sandbox/API:Stashedit

### CODE REVIEW

**Comment by zainababubakar** [comment]
Hi @brendajerop , kindly review

**Comment by zainababubakar** [comment]
Hi @brendajerop , while going through the module for this page ( https://en.wikipedia.org/w/api.php?action=help&modules=stashedit ), i saw this information below.
- Prepare an edit in shared cache.
- This is intended to be used via AJAX from the edit form to improve the performance of the page save.

While going through related pages for reference, i came across this page ( https://www.mediawiki.org/wiki/API:Edit/Editing_with_Ajax ).

Now having gone through all this, my question is, are these pages related ?, I'm a bit confused because there's not much for me to work with on the Stashedit page.

**Comment by brendajerop** [comment]
> Will deleting https://www.mediawiki.org/wiki/User:Zaycodes/Sandbox/API:Mergehistory affect https://www.mediawiki.org/wiki/User:Zaycodes/Sandbox/API:Stashedit
Yes, because the page you will be querying will be non-existent

I think I answered the rest of your questions on Zulip.

**Comment by zainababubakar** [comment]
Hi @brendajerop, kindly review, I've pushed changes

File: php/stashedit.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        "action" => "stashedit",
+        "title" => "PageTitle",
+        "section" => "new",
+        "sectiontitle" => "testing stashedit",
+        "text" => "testing stashedit API",
+        "contentmodel" => "text",
+        "contentformat" => "text/plain",
+        "baserevid" => "",
+        "format" => "json",
+		"token" => $csrftoken
```
Comment (brendajerop) [inline php/stashedit.php]: Indentation

File: python/stashedit.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+DATA = R.json()
+
+CSRF_TOKEN = DATA["query"]["tokens"]["csrftoken"]
+
+# Step 4: Send a POST request to prepare an edit in shared cache
+
+PARAMS_4 = {
+    "token":CSRF_TOKEN,
+    "action":"stashedit",
+    "title":"PageTitle",
```
Comment (brendajerop) [inline python/stashedit.py]: Let's use an actual page which has revision ids, say `Sandbox`:https://test.wikipedia.org/wiki/Sandbox

**Comment by zainababubakar** [comment]
Hi @brendajerop, I've made the changes

**Comment by brendajerop** [comment]
@zainababubakar Please rebase your changes on top of the new changes in master. There is a conflict :)

**Comment by zainababubakar** [comment]
@brendajerop i've resolved the conflicts

**Comment by brendajerop** [comment]
@zainababubakar  I've merged your code but made a few changes to clean up your git history since your code was essentially correct :)

**Comment by zainababubakar** [comment]
> 
> 
> @zainababubakar I've merged your code but made a few changes to clean up your git history since your code was essentially correct :)

Thanks @brendajerop

**Pull request merged**